src/mame/drivers/goupil.cpp

```diff
@@ -96,6 +98,34 @@ class goupil_g1_state : public driver_device
 	floppy_image_device *m_floppy;
 };
 
+ROM_START( visu_24x80 )
+	ROM_REGION( 0x800, "visu_24x80", 0 )
+	ROM_LOAD( "SMT_Goupil_G2_Charset_24x80.bin", 0x0000, 0x0800, CRC(F0F83B99) SHA1(75A7730AEC30280EE4CCF3DCAF587EEA4F861196) )
+ROM_END
+
+class goupil_g2_state : public goupil_g1_state
+{
+public:
+
+	goupil_g2_state(const machine_config &mconfig, device_type type, const char *tag)
+		: goupil_g1_state(mconfig, type, tag),
+		m_palette(*this, "palette"),
+		m_visu24x80_ram(*this, RAM_TAG),
+		m_visu24x80_rom(*this, "visu_24x80")
+	{
+	}
+
+	MC6845_UPDATE_ROW(crtc_update_row);
+	MC6845_ON_UPDATE_ADDR_CHANGED(crtc_update_addr_changed);
+
+	DECLARE_READ8_MEMBER(visu24x80_ram_r);
+	DECLARE_WRITE8_MEMBER(visu24x80_ram_w);
+
+	required_device<palette_device> m_palette;
```

Comment:
Is there a reason all these members need to be public?  If not, it's better to make them protected/private.